components/organisms/ProfileForm/ProfileForm.test.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { cleanup, render, fireEvent, wait } from '@testing-library/react';
+import 'jest-styled-components';
+import renderWithTheme from '__tests__/helpers/renderWithTheme';
+import * as actions from 'lib/apollo/hooks/actions';
+import ErrorDecorator from 'decorators/ErrorDecorator';
+
+import ProfileForm from './ProfileForm';
+
+jest.mock('lib/apollo/hooks/actions');
+jest.mock('decorators/ErrorDecorator');
+const originalUseUpdateUser = actions.useUpdateUser;
+
+describe('ProfileForm', () => {
+  afterEach(() => {
+    actions.useUpdateUser = originalUseUpdateUser;
+    jest.clearAllMocks();
+    jest.resetModules();
+    cleanup();
+  });
+
+  test('should call useUpdateUser on submit', async () => {
+    // Arrange
+    const expectedEmail = 'test@test.com';
+    const expectedFirstName = 'Test FirstName';
+    const expectedLastName = 'Test LastName';
+    const expectedButtonText = 'Update';
+    const expectedProfile = {
+      email: expectedEmail,
+      firstName: expectedFirstName,
+      lastName: expectedLastName,
+    };
+
+    const expectedValues = {
+      ...expectedProfile,
+      password: '',
+      currentPassword: '',
+      [expectedButtonText]: expectedButtonText,
+    };
+    const mockUpdateUser = jest.fn(() => Promise.resolve());
+    actions.useUpdateUser = jest.fn(() => [mockUpdateUser]);
+
+    const { getByText } = render(renderWithTheme(<ProfileForm profile={expectedProfile} />));
```

Comment:
Can we use `MockedProvider` from `@apollo/react-testing` as in Premiumparking? We can test with mutations in this case

Reply:
We know nothing about apollo in this component. I think we can use `MockedProvider` in the Profile page test